File: benchmarks/pandas/bench_dataframe_from2d_select.py
"""Benchmark: DataFrame from 2D array and column selection"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data2d = np.column_stack([
    np.arange(ROWS, dtype=float),
    np.arange(ROWS, dtype=float) * 2,
    np.arange(ROWS, dtype=float) * 3,
])
cols = ["a", "b", "c"]
df = pd.DataFrame(data2d, columns=cols)

for _ in range(WARMUP):
    pd.DataFrame(data2d, columns=cols)
    df[["a", "c"]]

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.DataFrame(data2d, columns=cols)
    df[["a", "c"]]
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "dataframe_from2d_select",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_dataframe_from2d_select.ts
/**
 * Benchmark: DataFrame.from2D and DataFrame.select
 */
import { DataFrame } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data2D = Array.from({ length: ROWS }, (_, i) => [i * 1.0, i * 2.0, i * 3.0]);
const cols = ["a", "b", "c"];
let df = DataFrame.from2D(data2D, cols);

for (let i = 0; i < WARMUP; i++) {
  DataFrame.from2D(data2D, cols);
  df.select(["a", "c"]);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  DataFrame.from2D(data2D, cols);
  df.select(["a", "c"]);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dataframe_from2d_select",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
